### PR TITLE
Support building ASP.NET Core projects with MSBuild if we have a VS solution

### PR DESCRIPTION
**Pull request by ahmelsayed**
~~This is bending getting msbuild to work with `set NUGET_PACKAGES=%HOME%\.nuget`~~

### CODE REVIEW

File: lib/generator._js [hunk 1]
```diff
@@ -264,10 +265,19 @@ ScriptGenerator.prototype.generateAspNetCoreDeploymentScript = function (_) {
     throw new Error('Only batch script files are supported for ASP.NET Core Application');
   }
 
+  if (!this.solutionPath && this.useMSBuild) {
+    throw new Error('Missing solution file path (--solutionFile). Solution file path has to be specified when used with --use-msbuild');
```
Comment (davidebbo) [inline lib/generator._js]: Is the converse also true? i.e. is it valid to pass solution when *not* passing `--use-msbuild`?

**Comment by davidebbo** [comment]
:shipit:

File: lib/generator._js (hunk 1 above)
Comment (ahmelsayed) [inline lib/generator._js]: If the caller doesn't pass `useMSBuild` but pass a solution file then the solution file will be ignored. Should we throw on that as well? I guess we can.

Though this remind me to update `--use-msbuild` to `--useMSBuild`

Comment (davidebbo) [inline lib/generator._js]: Well, alternative design would be to not need `--use-msbuild` at all, and just use the presence of the solution as a determinant. Makes for a simpler command line, and removes some invalid combinations that we need to check for,

Comment (ahmelsayed) [inline lib/generator._js]: Well now I feel stupid not having thought of that. Fixed this and kudu.

File: lib/generator._js [hunk 2]
```diff
@@ -266,8 +267,13 @@ ScriptGenerator.prototype.generateAspNetCoreDeploymentScript = function (_) {
 
   log.info('Generating deployment script for ASP.NET Core Application');
 
+  var msbuildArguments = '"%DEPLOYMENT_SOURCE%\\' + this.solutionPath + '" /nologo /verbosity:m /t:Build /p:AutoParameterizationWebConfigConnectionStrings=false;Configuration=Release;UseSharedCompilation=false';
```
Comment (davidebbo) [inline lib/generator._js]: So all those `/p` things apply in this Core scenario?

File: lib/generator._js [hunk 3]
```diff
@@ -266,8 +267,13 @@ ScriptGenerator.prototype.generateAspNetCoreDeploymentScript = function (_) {
...
+  msbuildArguments += ' %SCM_BUILD_ARGS%';
```
Comment (davidebbo) [inline lib/generator._js]: Why append this in a separate line?

**Pull request merged**